devtools/maven/src/main/java/io/quarkus/maven/TrackConfigChangesMojo.java

```diff
@@ -102,34 +112,64 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
             compareFile = recordedBuildConfigDirectory.toPath().resolve(this.recordedBuildConfigFile.toPath());
         }
 
-        final Properties compareProps = new Properties();
-        if (Files.exists(compareFile)) {
-            try (BufferedReader reader = Files.newBufferedReader(compareFile)) {
-                compareProps.load(reader);
-            } catch (IOException e) {
-                throw new RuntimeException("Failed to read " + compareFile, e);
-            }
-        } else if (!dumpCurrentWhenRecordedUnavailable) {
-            getLog().info(compareFile + " not found");
+        final boolean prevConfigExists = Files.exists(compareFile);
+        if (!prevConfigExists && !dumpCurrentWhenRecordedUnavailable && !dumpDependencies) {
+            getLog().info("Config dump from the previous build does not exist at " + compareFile);
             return;
         }
 
         CuratedApplication curatedApplication = null;
         QuarkusClassLoader deploymentClassLoader = null;
         final ClassLoader originalCl = Thread.currentThread().getContextClassLoader();
-        Properties actualProps;
         final boolean clearPackageTypeSystemProperty = setPackageTypeSystemPropertyIfNativeProfileEnabled();
         try {
             curatedApplication = bootstrapApplication(launchMode);
```

Comment:
FWIW, I benchmarked things a bit and resolving the dependencies is quite slow. Even for a relatively simple IT as the Hibernate Validator one, it takes ~ 0.8 s.

Reply:
It is, this is something i've been looking into for the last few weeks. But there is not much we can do especially for relatively simple projects, since most of the time is spent in the Maven artifact resolver itself.

However, the app model resolution will typically be necessary either way. It's about at which point it's performed. Here, we are moving this step to an earlier phase and it's cached for later phases if they get executed.